Enterprise-CMCS/managed-care-review
# [MCR-2614] backend changes for mccrsid

## Summary

This PR creates a new `updateContract` endpoint. This code is separate from the `formData` and is used to update the `mccrsID` field 

#### Related issues

https://qmacbis.atlassian.net/browse/MCR-2614

#### To Do

- [x] Fix the existing test for `updatingHealthPlanFormData`  that's failing (this may be an issue with the db refactor)
- [x] Add additional test for the `updateContract` postgres function
- [x] Error handling in the `updateContractResolver` 


#### Test cases covered

- Updating contracts is restricted to CMS users and the contract must be submitted 

## QA guidance

This can be tested with the graphQL Explorer http://localhost:3000/dev/graphql-explorer
- Add the `updateContract` mutation
```
mutation UpdateContract($input: UpdateContractInput!) {
  updateContract(input: $input) {
    pkg {
      id
      mccrsID
    }
  }
}
```

- Under variables supply an `id` that can be found from the network tab on the review submission page. You should also supply a `mccrsID` 
- You will need to be signed in as a CMS user
- Once run, you should see the `mccrsID` in the response
